mmv1/third_party/terraform/services/compute/resource_compute_network_test.go.erb

```diff
@@ -106,6 +107,43 @@ func TestAccComputeNetwork_routingModeAndUpdate(t *testing.T) {
 	})
 }
 
+func TestAccComputeNetwork_numericId(t *testing.T) {
+	t.Parallel()
+
+	var network compute.Network
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckComputeNetworkDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeNetwork_basic(acctest.RandString(t, 10)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeNetworkExists(
+						t, "google_compute_network.bar", &network),
+					testAccCheckComputeNetworkIsAutoSubnet(
+						t, "google_compute_network.bar", &network),
+				),
+			},
+			{
+				Config: testAccComputeNetwork_basic(acctest.RandString(t, 10)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeNumericId(
+						t, "google_compute_network.bar", &network),
```

Comment:
We could use either for checking `numeric_id` and `name` values but when it comes to checking the `id` there's a bit more required since there isn't already a function that does the resource format (`projects/{{project}}/global/networks/{{name}}`)

I mostly created the custom function for this and also so that it matches the style of the test file

Reply:
Note from pairing: `TestMatchResourceAttr` would be better for numeric_id!

Reply:
> Also, if you're refactoring the `testAccComputeNetwork_basic` function so you can specify the resource name using `networkName`, it would be good to refactor it so the function takes in a context map, similar to generated tests.

I had been working on refactoring the `testAccComputeNetwork_basic` to use `resource_name` instead of `suffix` as the input but it would be best to create a separate PR that refactors this. This would keep this PR focused on adding just the `numeric_id` field to `google_compute_network`.

Reply:
Yeah sure, I support that. Though we should do it soon to make sure it's not forgotten!